src/dbnode/client/aggregate_results_accumulator_consistency_test.go

```diff
@@ -0,0 +1,455 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package client
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/cluster/shard"
+	"github.com/m3db/m3/src/dbnode/generated/thrift/rpc"
+	"github.com/m3db/m3/src/dbnode/topology"
+	tu "github.com/m3db/m3/src/dbnode/topology/testutil"
+)
+
+var (
+	testStartTime, testEndTime   time.Time
+	testAggregateSuccessResponse = rpc.AggregateQueryRawResult_{}
+	errTestAggregate             = fmt.Errorf("random error")
+)
+
+func TestAggregateResultsAccumulatorAnyResponseShouldTerminateConsistencyLevelOneSimpleTopo(t *testing.T) {
+	// rf=3, 30 shards total; three identical hosts
+	topoMap := tu.MustNewTopologyMap(3, map[string][]shard.Shard{
+		"testhost0": tu.ShardsRange(0, 29, shard.Available),
+		"testhost1": tu.ShardsRange(0, 29, shard.Available),
+		"testhost2": tu.ShardsRange(0, 29, shard.Available),
+	})
+
+	// any response should satisfy consistency lvl one
+	for i := 0; i < 3; i++ {
+		testFetchTaggedWorkflow{
+			t:       t,
+			topoMap: topoMap,
+			level:   topology.ReadConsistencyLevelOne,
+			steps: []testFetchTaggedWorklowStep{
+				testFetchTaggedWorklowStep{
+					hostname:        fmt.Sprintf("testhost%d", i),
+					aggregateResult: &testAggregateSuccessResponse,
+					expectedDone:    true,
+				},
+			},
+		}.run()
+	}
+
+	// should terminate only after all failures, and say it failed
+	testFetchTaggedWorkflow{
```

Comment:
Nice.  Should we perhaps rename this to `fetchStateWorkflow` to more closely mimic the data structures it's testing (now that it's used for fetch tagged and aggregate)?

Reply:
sure thing

Reply:
done